gpiozero/output_devices.py

```diff
@@ -562,8 +562,12 @@ def __init__(
             raise GPIOPinMissing('red, green, and blue pins must be provided')
         LEDClass = PWMLED if pwm else LED
         super(RGBLED, self).__init__()
-        self._leds = tuple(LEDClass(pin, active_high) for pin in (red, green, blue))
-        self.value = initial_value
+        try:
+            self._leds = tuple(LEDClass(pin, active_high) for pin in (red, green, blue))
+            self.value = initial_value
+        except:
+            self.close()
+            raise
```

Comment:
Ooops - yup, I'll chop that out in a mo - just wrestling with getting weak-method refs working in py2 (stupid errors on my part in the WeakMethod backport).